ir/memory.cpp

```diff
@@ -463,12 +463,28 @@ Memory::Memory(State &state, bool little_endian)
                              avail_space);
   }
 
-  // Initialize a memory block for null pointer.
-  // TODO: in twin memory model, this is not needed.
   if (state.isSource()) {
+    // Initialize a memory block for null pointer.
+    // TODO: in twin memory model, this is not needed.
     auto nullPtr = Pointer::mkNullPointer(*this);
     state.addAxiom(nullPtr.get_address(false) == 0);
     state.addAxiom(nullPtr.block_size(false) == 0);
+
+    // Non-local blocks are disjoint.
+    // Ignore null pointer block
+    for (unsigned bid = 1; bid <= non_local_bid_upperbound; ++bid) {
+      Pointer p1(*this, bid, false);
+      for (unsigned bid2 = bid + 1; bid2 <= non_local_bid_upperbound; ++bid2) {
+        Pointer p2(*this, bid2, false);
+        state.addAxiom((p1.is_block_alive() && p2.is_block_alive())
+            .implies(disjoint(p1.get_address(), p1.block_size(),
+                              p2.get_address(), p2.block_size())));
+      }
+      // Non-local blocks' addr + size does not overflow.
+      state.addAxiom(p1.get_address().add_no_uoverflow(p1.block_size()));
+      // Size is less than half of memory
+      state.addAxiom(p1.block_size().extract(bits_size_t-1, bits_size_t-1) == 0);
```

Comment:
this is not needed. It's already given for free. See the definition of `block_size()`